Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp

```diff
@@ -281,9 +289,19 @@ void ThreadedCompositor::renderLayerTree()
     // GL viewport is updated separately, if necessary. This establishes sequencing where
     // everything inside the will-render and did-render scope is done for a constant-sized scene,
     // and similarly all GL operations are done inside that specific scope.
-
+#if !ENABLE(RESIZE_WAYLAND_SURFACES_ON_SUSPEND_PAINTING)
     if (needsResize)
         m_client.resize(viewportSize);
+#else
+    if (needsResize && (m_nonCompositedWebGLEnabled || m_suspendToTransparentState != SuspendToTransparentState::Requested)) {
```

Comment:
You don't need to check whether m_nonCompositedWebGL is enabled here, cause in order to execute this code it must be disabled. Check that the first thing this function does is checking whether it's enabled, to call renderNonCompositedWebGL() and then return, so we know it's disabled in the rest of the function.